File: tests_wraps/src/main/java/com/github/klee0kai/test/boxed/model/CarBoxedInjectLists.java
package com.github.klee0kai.test.boxed.model;

import com.github.klee0kai.test.car.model.Bumper;
import com.github.klee0kai.test.car.model.Wheel;
import com.github.klee0kai.test.car.model.Window;

import javax.inject.Inject;
import java.util.List;

public class CarBoxedInjectLists {


    @Inject
    public List<CarBox<Bumper>> bumpers;

    @Inject
    public List<CarBox<Wheel>> wheels;

    @Inject
    public List<CarBox<Window>> windows;

    public List<CarBox<Bumper>> bumpersMethodFrom;

    public List<CarBox<Wheel>> wheelsMethodFrom;

    public List<CarBox<Window>> windowsMethodFrom;

    @Inject
    public void init(List<CarBox<Bumper>> bumpers, List<CarBox<Wheel>> wheels, List<CarBox<Window>> windows) {
        bumpersMethodFrom = bumpers;
        wheelsMethodFrom = wheels;
        windowsMethodFrom = windows;
    }


}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/boxed/model/CarBoxedInjectLists.kt
package com.github.klee0kai.test.boxed.model

import com.github.klee0kai.stone.annotations.qualifier.IgnoreQualifier
import com.github.klee0kai.test.car.model.Bumper
import com.github.klee0kai.test.car.model.Wheel
import com.github.klee0kai.test.car.model.Window
import javax.inject.Inject

class CarBoxedInjectLists {

    @Inject
    @IgnoreQualifier
    var bumpers: List<CarBox<Bumper>>? = null

    @Inject
    @IgnoreQualifier
    var wheels: List<CarBox<Wheel>>? = null

    @Inject
    @IgnoreQualifier
    var windows: List<CarBox<Window>>? = null

    var bumpersMethodFrom: List<CarBox<Bumper>>? = null

    var wheelsMethodFrom: List<CarBox<Wheel>>? = null

    var windowsMethodFrom: List<CarBox<Window>>? = null


    @Inject
    fun init(
        @IgnoreQualifier
        bumpers: List<CarBox<Bumper>>?,

        @IgnoreQualifier
        wheels: List<CarBox<Wheel>>?,

        @IgnoreQualifier
        windows: List<CarBox<Window>>?,
    ) {
        bumpersMethodFrom = bumpers
        wheelsMethodFrom = wheels
        windowsMethodFrom = windows
    }
}
